test/basics/aligned_readXX_basic.c

```diff
@@ -0,0 +1,60 @@
+#ifdef HAVE_CONFIG_H
+# include <config.h>
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <unistd.h>
+#include <qthread/qthread.h>
+#include "argparsing.h"
+
+// Test that a readXX on a full var performs the read, and leaves the FEB state
+// unchanged
+static void testReadXXOnFull(void)
+{
+    aligned_t x, val;
+
+    x = 45, val = 55;
+    assert(qthread_feb_status(&x) == 1);
+
+    iprintf("Before x=%d, val=%d, x_full=%d\n", x, val, qthread_feb_status(&x));
+    qthread_readXX(&val, &x);
+    iprintf("After  x=%d, val=%d, x_full=%d\n", x, val, qthread_feb_status(&x));
+
+    assert(qthread_feb_status(&x) == 1);
+    assert(x == 45);
+    assert(val == 45);
+}
+
+// Test that a readXX on an empty var performs the read, and leaves the FEB
+// state unchanged
+static void testReadXXOnEmpty(void)
+{
+    aligned_t x, val;
+
+    x = 45, val = 55;
+    qthread_empty(&x);
+
+    iprintf("Before x=%d, val=%d, x_full=%d\n", x, val, qthread_feb_status(&x));
+    qthread_readXX(&val, &x);
+    iprintf("After  x=%d, val=%d, x_full=%d\n", x, val, qthread_feb_status(&x));
+
+    assert(qthread_feb_status(&x) == 0);
+    assert(x == 45);
+    assert(val == 45);
```

Comment:
I just updated https://github.com/Qthreads/qthreads/pull/25 with the details, but Chapel expects/needs empty variables to have the same value they had when they were full. Assuming qthreads is also willing to do that too, I think this assert should stay, otherwise we probably can't use native qthread sync vars.

And note that the current qthreads implementation does preserve the value, or rather it doesn't go out of it's way anywhere to invalidate it.